router/usecase/optimized_routes.go

```diff
@@ -61,6 +50,19 @@ func estimateAndRankSingleRouteQuote(ctx context.Context, routes []route.RouteIm
 
 	// If we skipped all routes due to errors, return the first error
 	if len(routesWithAmountOut) == 0 && len(errors) > 0 {
+		// If we encounter this problem, we attempte to invalidate all caches to recompute the routes
+		// completely.
+		// This might be helpful in alloyed cases where the pool gets imbalanced and runs out of liquidity.
+		// If the original routes were computed only through the zero liquidity token, they will be recomputed
+		// through another token due to changed order.
+
+		// Note: the zero length check occurred at the start of function.
+		tokenOutDenom := routes[0].GetTokenOutDenom()
+
+		r.candidateRouteCache.Delete(formatCandidateRouteCacheKey(tokenIn.Denom, tokenOutDenom))
+		tokenInOrderOfMagnitude := GetPrecomputeOrderOfMagnitude(tokenIn.Amount)
+		r.rankedRouteCache.Delete(formatRankedRouteCacheKey(tokenIn.Denom, tokenOutDenom, tokenInOrderOfMagnitude))
```

Comment:
I assume there is some logic somewhere that repopulates if the cache is empty

Reply:
Yes, since cache is empty, it would try to recompute the routes and fill in the cache with the new routes